src/traits.jl

```diff
@@ -58,3 +58,33 @@ Return the [`OutputTrait`] of the [`Model`](@ref) or [`Template`](@ref).
 """
 output_type(::T) where T = output_type(T)
 output_type(T::Type) = throw(MethodError(output_type, (T,)))  # to prevent recursion
+
+"""
+    InputTrait
+
+The `InputTrait` specifies if the model supports point or distribution inputs to predict,
+denoted by [`PointInput`](@ref) or [`DistributionInput`](@ref), respectively.
+"""
+abstract type InputTrait end
+
+"""
+    PointInput <: InputTrait
+
+Specifies that the [`Model`](@ref) accepts real-valued input variables to `predict`.
+"""
+abstract type PointInput <: InputTrait end
+
+"""
+    DistributionInput <: InputTrait
+
+Specifies that the [`Model`](@ref) accepts a distribution over the input variables to `predict`.
+"""
+abstract type DistributionInput <: InputTrait end
+
+"""
+    input_type(::T) where T = input_type(T)
+
+Return the [`InputTrait`] of the [`Model`](@ref) or [`Template`](@ref).
+"""
+input_type(::T) where T = input_type(T)
```

Comment:
Would it make things non-breaking if we defined
```julia
input_type(::Type{<:Model}) = PointInput
input_type(::Type{<:Template}) = PointInput
```
then models could overload this if needed? (Since at the moment we probably assume everything is `PointInput`)